paddle/fluid/framework/reader.cc

```diff
@@ -13,11 +13,39 @@
 // limitations under the License.
 
 #include "paddle/fluid/framework/reader.h"
+#include <deque>
 
 namespace paddle {
 namespace framework {
 ReaderBase::~ReaderBase() {}
 
+void ReaderBase::InsertDecoratedReader(
+    const std::shared_ptr<ReaderBase> &decorated_reader) {
```

Comment:
Here, it should be better to check the type of `decorated_reader` is exactly `DecoratedReader`.